Automattic/jetpack
# VideoPress: Use playback tokens to get resources when the VideoPress site privacy is set to private

Fixes #27314, #26938 and #27391.

#### Changes proposed in this Pull Request:

* Renamed the `videoIsPrivate` flag to `needsPlaybackToken`
* Changed the resolver to rely on the new `needsPlaybackToken` flag when deciding if a video needs a playback token
* Exposed the `needsPlaybackToken` flag for all the videos on the client initial state

#### Other information:


#### Jetpack product discussion



N/A.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Check the "Video Privacy: Restrict views to members of this site" setting:

<img width="408" alt="Screen Shot 2022-11-11 at 19 16 20" src="https://user-images.githubusercontent.com/6760046/201438389-1d52d728-9440-4cbe-ad57-37b702ae7d36.png">

* Upload some videos to the media library, leaving the privacy setting as default
* Open the browser console to inspect errors and the network activity
* Do a hard refresh on the page
* Before the fix, the thumbnails for the added videos were broken:

<img width="1140" alt="Screen Shot 2022-11-11 at 19 32 20" src="https://user-images.githubusercontent.com/6760046/201439907-3f41fcda-1f29-45d8-a97b-4516041ed3d6.png">

* After the fix, you should see all the thumbnais properly:

<img width="1112" alt="Screen Shot 2022-11-11 at 19 22 52" src="https://user-images.githubusercontent.com/6760046/201439104-61a68292-8815-47d2-bf3a-8311657f525b.png">

* Before the fix, you would see failed `403` responses for the thumbnail files:

<img width="755" alt="Screen Shot 2022-11-11 at 19 33 18" src="https://user-images.githubusercontent.com/6760046/201439925-91edfda2-acda-47fc-819d-3756076cdbd1.png">

* After the fix, no `403` responses should appear on the console
* After the fix, you also will see requests to the playback token endpoint on the network log, showing that the application is now requesting the tokens that it will need:

<img width="597" alt="Screen Shot 2022-11-11 at 19 26 15" src="https://user-images.githubusercontent.com/6760046/201439392-f3b8faa2-8e67-4181-ad16-7d90decdb9b4.png">

* Make sure to test the list view as well as the grid view
